api/controllers/site.js

```diff
@@ -68,7 +68,7 @@ module.exports = {
         siteJSON = json;
         return authorizer.destroy(req.user, site);
       })
-      .then(() => S3SiteRemover.removeSite(site))
+      .then(() => S3SiteRemover.removeSite(site, req.user))
```

Comment:
What are your thoughts on moving the GH check to the authorizer? That way the `S3SiteRemove` stays focused on removing sites, and the policies for whether someone is authorized to remove a site are colocated.